File: backend/Models/Medication.cs
using System.ComponentModel.DataAnnotations;
namespace backend.Models
{
    public class Medication
    {
        [Key]
        public int Id {get; set;}
        [Required]
        public string Name {get; set;}= string.Empty;
        [Required]
        public string? Description {get; set;}

        public string? Type {get; set;}
        //pills, syrup or cream

        public string? SideEffects{get; set;}
    }
}

File: front end/Flutter/lib/models/medication.dart
class Medication {
  final int? Id;
  final String name;
  final String type;
  final String description;
  final String? sideEffects;
  Medication({
    required this.Id,
    required this.name,
    required this.type,
    required this.description,
    required this.sideEffects,
  });
  factory Medication.fromJson(Map<String,dynamic> json){
    return Medication(
      Id: json['id'],
      name: json['name'],
      type: json['type'],
      description: json['description'],
      sideEffects: json['sideEffects'],

    );
  }
  Map<String, dynamic> toJson(){
    return{
      'name': name,
      'description': description,
      'type': type,
      'sideEffects': sideEffects,
    };
  }
}